internal/api/mfa.go

```diff
@@ -259,12 +287,24 @@ func (a *API) VerifyFactor(w http.ResponseWriter, r *http.Request) error {
 			return forbiddenError(ErrorCodeMFAVerificationRejected, output.Message)
 		}
 	}
+
 	if !valid {
+		if shouldReEncrypt && config.Security.DBEncryption.Encrypt {
+			es, err := crypto.NewEncryptedString(factor.ID.String(), []byte(secret), config.Security.DBEncryption.EncryptionKeyID, config.Security.DBEncryption.EncryptionKey)
```

Comment:
at first glance this seems similar to the function below. Is it possible to extract into a function that semantically makes sense?

Reply:
Maybe, I'll see.

Reply:
Tried to hide the repetition as much as I could.